### PR TITLE
[icalendar] Change migration of last_update to use ThingHandlerCallback

### LINKED ISSUE #9674: [icalendar] Channel for last update is not added correctly while migration
In merge request #9616, the last_update channel migration was added. In the discussion after merging, @cweitkamp pointed to the wrong creation of the channel which should be fixed.

### PR DESCRIPTION
**Pull request by daMihe**
Uses now also the `ThingHandlerCallback`'s factory of `ChannelBuilder`s for creating the channel on old instances.

Closes #9674

### CODE REVIEW

File: bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java [hunk 1]
```diff
@@ -269,12 +270,18 @@ private void executeEventCommands(List<Event> events, CommandTagType execTime) {
...
      */
     private void migrateLastUpdateChannel() {
-        Thing thing = getThing();
+        final Thing thing = getThing();
         if (thing.getChannel(CHANNEL_LAST_UPDATE) == null) {
             logger.debug("last_update channel is missing in this Thing. Adding it.");
-            ThingBuilder thingBuilder = editThing();
-            ChannelBuilder channelBuilder = ChannelBuilder.create(new ChannelUID(thing.getUID(), CHANNEL_LAST_UPDATE));
-            thingBuilder.withChannel(channelBuilder.withType(LAST_UPDATE_TYPE_UID).build());
+            final ThingBuilder thingBuilder = editThing();
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java:276]: One minor improvement: You can call this method shortly before `withChannel(...)`.

**Review by cweitkamp — COMMENTED** [review]
Thanks for the changes. One minor comment.

File: bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java (hunk 1 above)
Comment (daMihe) [reply bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java]: Yep. The ChannelBuilder can also be final. Done.

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Comment by daMihe** [comment]
> LGTM. Thanks.

Thank You for your patience 🙂

**Comment by cweitkamp** [comment]
> Thank You for your patience

I have to thank you to find the time to implement all this.

**Pull request merged**